tests/components/ecobee/test_climate.py

```diff
@@ -76,6 +77,17 @@ async def test_name(thermostat):
     assert thermostat.name == "Ecobee"
 
 
+async def test_aux_heat_not_supported_by_default(ecobee_fixture, thermostat):
```

Comment:
I don't have the time to completely refactor the tests right now.  Please feel free to take this over.  Its a shame though to have the aux functionality completely broken due to this.  Right now it is hardcoded as supported.

Reply:
If I do find time later, what would be required here?

1. completely refactor `test_climate.py` to follow the new standards
2. Make a second file with just the new tests I added following the new standards
3. Mix both styles in the same file?

I have very limited python experience so any guidance is appreciated.

Reply:
It is not realistic to apply standards retroactively.

Its fine to have a mix of both until them until someone has time to re-write the legacy ones.

Reply:
Gotcha.  Any tips or guidance on how to customize the ecobee configuration for a teast?  

Calling `async_setup_component` directly for the climate platform results in an error that it is not supported.

I've been experimenting with [this setup_platform helper](https://github.com/home-assistant/core/blob/dev/tests/components/ecobee/common.py#L11), modified with an additional argument to pass custom data, but it appears to use the [fixture  data](https://github.com/home-assistant/core/blob/dev/tests/components/ecobee/fixtures/ecobee-data.json) no matter what data I  pass in

Reply:
Are you still stuck on this one, or good to go?

Reply:
No, I tried for a few hours, but have not been  able to figure out the right `patch` to be able to mock the configuration data.  Unfortunately it's going to be at least a week until I can work on this again.

If anyone wants to take over to finish out those tests, that would be appreciated.  
Otherwise any tips to try next week?